lib/dao.js

```diff
@@ -840,6 +840,19 @@ DataAccessObject.replaceOrCreate = function replaceOrCreate(data, options, cb) {
     return this.create(data, options, cb);
   }
 
+  var forceId = this.settings.forceId;
+  if (forceId) {
+    this.findById(id, function(err, instance) {
+      if (err || !instance) {
+        var msg = g.f('Cannot find unknown {{id}} "%s". ', id);
+        var error = new Error(msg);
+        error.statusCode = error.status = 404;
+        return cb(error);
+      }
+    });
+    return Model.replaceById(id, data, options, cb);
```

Comment:
Nope, that doesn't work .. I have tried it before and it doesn't work. The only way it works is the way the code is currently written